pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -869,7 +873,9 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
     // server returns STRING as default dataType for all columns in (some) scenarios where no rows are returned
     // this is an attempt to return more faithful information based on other sources
     if (brokerResponse.getNumRowsResultSet() == 0) {
```

Comment:
Shouldn't query option override default ?
With current implementation it doesn't seem possible to override `_useMSQEWhenEmptySchema=true` .

Reply:
You're right, I rushed it and the expected behavior is as you described.

578fea9546